Firestore/core/src/firebase/firestore/remote/datastore.mm

```diff
@@ -148,23 +148,23 @@ void LogGrpcCallFinished(absl::string_view rpc_name,
 
 void Datastore::CommitMutations(NSArray<FSTMutation*>* mutations,
                                 FSTVoidErrorBlock completion) {
-  grpc::ByteBuffer message = serializer_bridge_.ToByteBuffer(
-      serializer_bridge_.CreateCommitRequest(mutations));
-
   ResumeRpcWithCredentials(
-      [this, message, completion](const StatusOr<Token>& maybe_credentials) {
+      [this, mutations, completion](const StatusOr<Token>& maybe_credentials) {
```

Comment:
I'm not sure how this works with objc++...

Your lambda is capturing the mutations pointer and then using it later. Is it possible that the thing the pointer points to becomes invalid and/or something else by the time the lambda is actually executed?

Reply:
My understanding is that ARC makes all Objective-C pointers behave similarly to `std::shared_ptr`.

Reply:
So it should still exist due to arc, though may have been mutated. But I think the FSTMutation instances are (somewhat ironically) immutable. And I don't believe we change the array itself either, so this seems reasonable.

It may be worth a quick docstring in the header to indicate that the nsarray shouldn't be changed?

Reply:
In the Objective-C style guide it's specifically called out that if you're going to retain an `NSString` for a while you should make your own copy, e.g. via `[foo copy]`. If `foo` is actually an `NSMutableString`, this will create a copy, but otherwise this is a no-op.

This advice does not extend to other types, even if they implement `NSCopying`. Conventionally for things like arrays, it's assumed that if you're giving away your pointer to it, it's no longer modifiable.

If you want to be particularly careful about this you could make this take a `std::vector<FSTMutation*>&&` which would require the caller to explicitly give the mutations away. We eventually want to move to C++ anyway so moving to C++ containers before their contents can be moved is reasonable.

In any case, I don't think it's worth stressing about this. Documentation on this point isn't going to be read and the defensive copy is more expensive than it's worth.

Reply:
Just throw in my 1 cent. As far as this works for iOS client, I am fine with this As Is. Whether CMake build actually enabled arc for object c object I am not sure; but if the goal is to make this work for the iOS client SDK, I'd make this logic as simple as possible, which is to rely on the ARC as most of our current iOS object c code does. Later when we are ready to deploy the code to other platform, we can revisited this (actually, then the mutation will be in C++ instead of being FSTMutation, and thus there may not be any issue by then).

Reply:
@wilhuff Thanks for explaining.

@rsgowman All in all, I'm inclined to consider this not worth handling explicitly (this function only has one caller, after all), but I don't feel strongly about it. I can add a comment if you prefer.

Reply:
I'm fine with it as is. PR still lgtm.